sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,60 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([keys/0]).
```

Comment:
we may need the ability to define multiple scehmas per module
e.g. for nested sub ojbect.
```
-export([names/0, mappings/0, fields/1]).

names() -> [foo, nested].

mappings(foo) -> app_foo;
mappings(_) -> undefined.

fields(foo) -> [{"setting", fun setting/1}, 
{"nested", fun nested/1}
];
fields(nested) ->
[ {"abc", fun abc/1}
].

```

Reply:
with mapping defined at 'object' level, there is no need to define mappings for each field.